medusa/restore_node.py

```diff
@@ -195,7 +196,7 @@ def invoke_sstableloader(config, download_dir, keep_auth, fqtns_to_restore, stor
                     cql_username = 'foo' if config.cassandra.cql_username is None else config.cassandra.cql_username
                     cql_password = 'foo' if config.cassandra.cql_password is None else config.cassandra.cql_password
                     sstableloader_args = [config.cassandra.sstableloader_bin,
-                                          '-d', hostname_resolver.resolve_fqdn() if cassandra_is_ccm == 0
+                                          '-d', hostname_resolver.resolve_fqdn(socket.gethostbyname(socket.getfqdn())) if cassandra_is_ccm == 0
```

Comment:
Maybe we could avoid some duplication by coding this logic into `hostname_resolver.resolve_fqdn()` directly?

We could revert to making the ip_address optional again (and empty by default) and then use `socket.gethostbyname(socket.getfqdn())` if it's indeed empty in the method implementation.

it would give something like: 

```
    def resolve_fqdn(self, ip_address=''):
        ip_address_to_resolve = ip_address if ip_address != '' else socket.gethostbyname(socket.getfqdn())
        if str(self.resolve_addresses) == "False":
            logging.debug("Not resolving {} as requested".format(ip_address_to_resolve))
            return ip_address_to_resolve
        fqdn = socket.getfqdn(ip_address_to_resolve)
        logging.debug("Resolved {} to {}".format(ip_address_to_resolve, fqdn))
        return fqdn
```

wdyt?

Reply:
Why not yes, to be honest I found this `hostname_resolver.resolve_fqdn(socket.gethostbyname(socket.getfqdn()))` not really nice :p

Reply:
@adejanovski fix pushed 👌